openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -849,6 +861,9 @@ private void extractedConstraintAnnotation(MetadataNode metadata,
                                     if (kind == SyntaxKind.NUMERIC_LITERAL) {
                                         fillConstraintValue(constraintBuilder, name, expressionNode
                                                                     .toString().trim());
+                                    } else if (kind == SyntaxKind.REGEX_TEMPLATE_EXPRESSION) {
+                                        fillConstraintValue(constraintBuilder, name, expressionNode
+                                                .toString().replaceAll("re |`", "").trim());
```

Comment:
Shall we improve this value extract logic incase to avoid replacing some necessary backticks as well? As an example to address the regex pattern like `` ^[`a-zA-Z0-9_]+$ `` which backtick `` ` `` has the middle of the regex.

Reply:
Isn't there a specific node kind to get the evaluated regex string? If there is a lang function it is better to depend on that than having our own logic :)

Reply:
Yeah, it will be better to depend on the lang API for evaluating regex patterns, We will be looking into it.

Reply:
@SachinAkash01 Could you please let us know whether there is a way we can access only the regex string for a given expression node?

Reply:
Had a discussion with @LakshanWeerasinghe and @SachinAkash01 about this and came to conclusion that we can use template expression node for this task.

Reply:
we can use `(^\\)${` regex for check interpolations